Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -433,6 +442,19 @@ class Gx2Fitter {
         return;
       }
 
+      if (startVolume != nullptr &&
+          startVolume != state.navigation.startVolume) {
```

Comment:
I don't think we should couple to the navigation state here but rather check if the updated position is either too far away in 3D/4D global space/spacetime or if the update is still on check if the 3D global position is still in the same tracking volume as the initial parameters were

Reply:
For this we would need to do an analysis, on different detectors. Maybe we could get the constraints in as some extra configuration parameters of the fitter. Not completely sure yet how to do it, because it sounds like we need some kind of map for it. Starting parameters close to the beam pipe edge are more constraint (at least in one direction), than the ones in the center. Also the constraints depend on the direction itself.

Another idea would be to use the covariance to give constraints to the parameters.

However, I would prefer to look at it, after I get the material in. Maybe that already improves /solves it.

Reply:
IMO in principle this should never happen if the fitter works correctly so this is a workaround anyways for now. Having a cut parameter that can be configured does not sound too bad to me and generalizes the solution further than just checking the volume.

Anyways let's see what the material does.

In general this is blocking Gen1 navigation changes because I get errors which originate from the fitters and we monitor the faulty behavior in the CI. If we cannot get it fixed soon I would argue to disable physmon for the time being.

Reply:
In theory, we could get this in as it is. The only thing missing here is the unit-test, which is a bit weird. I already put in a few hours, but I couldn't construct the volume change yet.

I would try to flip the whole gx2f-tests but maybe if we want to explore different approaches, we could leave it for now?

Reply:
I don't have permissions to overwrite sonarcloud and the macos build is broken right now. Otherwise we can put this in from my side